Add saving, ui, and syntax settings to REPL

gseastream opened the pull request:
# Summary

I've used a small custom sublime repl plugin for python for years, and after discovering this package I think it'd be nice to include the couple of neat things I've enjoyed with my repl setup. Namely, saving the file before opening the repl, opening the repl in a second row tab below the file, and closing any still open repls in that second row (if they haven't been moved to the main row/window, in which case they're kept open). I made these features optional by putting them into settings, and I defaulted them to true since I think they're quite handy, but they can default to false if wanted.

I also added a setting for the syntax highlighting of the repl, since I noticed that it is currently set to python syntax. This is not common for python repls, which are usually highlighted as plain text, since python highlighting causes text in the repl to show up weird. I made the setting default to plain text to reflect this, leaving the option to still use python syntax, but again the default could be changed back. Note that without adding this setting the only way to change the highlighting is to change the global SublimeREPL setting (as far as I know).

# Changes

- Setting `repl_save_dirty` to save the file when opening the repl
- Setting `repl_nice_ui` to open the repl in a second row tab
- Setting `repl_syntax` to control syntax highlighting, defaulted to plain text

gseastream committed:
Add repl_syntax setting

gseastream committed:
Add nice ui settings for repl

mandeep commented:
Hi @gseastream thanks for the PR! I'll take a look at the changes on my machine and will get back to you with any questions.

mandeep commented:
@gseastream I had a chance to review your changes and I have a few remarks:

- Could you change the repl layout to be false by default?
- Would it be possible to rename the `repl_nice_ui` variable to something more descriptive? Maybe something like `repl_open_row` or `repl_open_below`. Something that the user will be able to decipher in case they don't read the comments.
- When closing the last tab in a repl that is in row format, the second row stays open and the user has to click View->Layout and change back to what they had before. Would it be possible to close this row when there are no tabs open in it?

Thanks again for your contribution. I think your contributions are really helpful!

gseastream committed:
Change name from repl_nice_ui to repl_open_row

gseastream committed:
Default repl_open_row to false

gseastream committed:
Add repl command to readme, with link to command names

gseastream committed:
Remove new row when closing repl tab

gseastream commented:
Thanks for the feedback! I added all your requests. This is actually the first plugin I've done real work on, so I've never even thought of automating the closing of the additional row, but after testing it out I quite enjoy it. There's also the possibility I did something wrong since I'm new to this so let me know.

I also added info to the readme about the repl command. And I threw in a reference to the file that lists all the command names since I struggled to find that when trying to bind the repl command to ctrl+b (my usual keystroke for building into repl).

gseastream committed:
Add comment about getting window during init

mandeep commented:
@gseastream thanks so much for the changes. Everything looks great and I'll release a new version of the plugin later today.

Thanks again!
